lib/assets/javascripts/cartodb/models/tabledata.js

```diff
@@ -733,18 +743,34 @@ cdb.admin.CartoDBTableData = cdb.ui.common.TableData.extend({
     }
 
     // use get to fetch the schema, probably cached
-    this._sqlQuery(_.template('select * from (<%= sql %>) __wrapped limit 0')({ sql: sql }), function(data) {
-      // get schema
-      self.query_schema = self._schemaFromQueryFields(data.fields);
-      if (!self.table.isInSQLView()) {
-        if ('the_geom' in self.query_schema) {
-          delete self.query_schema['the_geom_webmercator'];
+    //TODO This is a hacky way to make sure the table data returns before the get schema call.
+    //we need the table schema_alias to before normal schema to inject the alias into the columns
+    var self = this;
+    setTimeout(function(){
```

Comment:
I am not in front of the computer I can review but "is not related" So somewhere the table metadata is being called "can't remember where right now" and so one suggestion might be to see if schema_alias is undefined to then to figure out how to make that call and on that callback finish this _fetch function.  This way we could remove the setTimeout and make a call only if it was undefined which would be that it happened too early.  

Not sure any other way since these 2 calls are made never together to chain.  If you think that is okay I can try to sign on today and figure out how to call the table request.